plugins/default-browser-emulator/injected-scripts/SharedWorker.prototype.ts

```diff
@@ -1,20 +1,73 @@
-declare let newDocumentScript;
-declare let newDocumentScriptWrapper;
-
-if (typeof SharedWorker !== 'undefined') {
-  // shared workers don't automatically pause in devtools, so we have to manipulate
-  proxyConstructor(self, 'SharedWorker', (target, argArray) => {
-    if (!argArray?.length) return ProxyOverride.callOriginal;
-    const [url] = argArray;
-    if (url?.toString().startsWith('blob:')) {
-      // read blob contents synchronously
-      const xhr = new XMLHttpRequest();
-      xhr.open('GET', url, false);
-      xhr.send();
-      const text = xhr.response;
-      const newBlob = new Blob([`(${newDocumentScriptWrapper.toString()})();\n\n`, text]);
-      return ReflectCached.construct(target, [URL.createObjectURL(newBlob)]);
-    }
-    return ProxyOverride.callOriginal;
-  });
+
+// When launching a SharedWorker from blob it is possible that it launches so fast
+// that setting debugger breakpoint is too slow and not triggered. We handle this in two ways:
+// 1. SharedWorker blob is too fast -> create promise in blob and wait for injected logic to trigger promise resolver
+// 2. Debugger is fast enough -> set bool = true in injected logic -> sharedworker sees this and doesn't wait
+
+
+// Random variable so external scripts can't predict this
+const waiter = `wait-${sourceUrl}`;
+
+if (targetType === 'shared_worker') {
+  if (waiter in self) {
+    // trigger promise
+    self[waiter]();
+  } else {
+    // injected script fast enough -> no wait needed
+    self[waiter] = true;
+  }
 }
+
+if (typeof SharedWorker === 'undefined') {
```

Comment:
should this come before the waiter? Not sure if we need this clause or not, but if it's possible, it seems that we would be stranding the wait- variables